TCAT/Cells/ServiceAlertTableViewCell.swift

```diff
@@ -0,0 +1,238 @@
+                make.height.equalTo(timeSpanLabel.intrinsicContentSize.height).labeled("timeSpanLabel: Height")
+            }
+        } else {
+            timeSpanLabel.snp.remakeConstraints { (make) in
+                make.top.leading.trailing.equalToSuperview().inset(borderInset).labeled("timeSpanLabel: Top, Leading, Trailing")
+                make.height.equalTo(timeSpanLabel.intrinsicContentSize.height).labeled("timeSpanLabel: Height")
+            }
+        }
+        
+        descriptionLabel.snp.remakeConstraints { (make) in
+            make.top.equalTo(timeSpanLabel.snp.bottom).offset(12).labeled("descriptionLabel: Top")
+            make.leading.equalTo(timeSpanLabel).labeled("descriptionLabel: Leading")
+            make.trailing.equalToSuperview().inset(borderInset).labeled("descriptionLabel: Trailing")
+            if let text = descriptionLabel.text {
+                
+                let width = contentView.frame.width - (CGFloat)(2 * borderInset)
+                
+                let heightValue = ceil(text.heightWithConstrainedWidth(width: width, font: descriptionLabel.font))
+                make.height.equalTo(ceil(heightValue)).labeled("descriptionLabel: Height")
+            } else {
+                make.height.equalTo(descriptionLabel.intrinsicContentSize.height).labeled("descriptionLabel: Height")
+            }
+        }
+        
+        if let stackView = affectedRoutesStackView {
+            // When both a separator view and stackView are required
+            affectedRoutesLabel.snp.remakeConstraints { (make) in
+                make.leading.equalTo(timeSpanLabel).labeled("AffectedRoutesLabel: Leading")
+                make.top.equalTo(descriptionLabel.snp.bottom).offset(24).labeled("affectedRoutesLabel: Top")
+                make.width.equalTo(affectedRoutesLabel.intrinsicContentSize.width).labeled("affectedRoutesLabel: Width")
+                make.height.equalTo(affectedRoutesLabel.intrinsicContentSize.height).labeled("affectedRoutesLabel: Height")
+            }
+            
+            stackView.snp.remakeConstraints { (make) in
+                make.top.equalTo(affectedRoutesLabel.snp.bottom).offset(8).labeled("affectedRoutesStackView: Top")
+                make.leading.equalTo(timeSpanLabel).labeled("affectedRoutesStackView: Leading")
+                make.trailing.bottom.equalToSuperview().inset(borderInset).labeled("affectedRoutesStackView: Trailing, Bottom")
+            }
+            
+        } else {
+            descriptionLabel.snp.makeConstraints { (make) in
+                make.bottom.equalToSuperview().inset(borderInset).labeled("descriptionLabel: Bottom")
+            }
+        }
+        super.updateConstraints()
+    }
+    
+    private func formatTimeString(_ fromDate: String, toDate: String) -> String {
+        
+        // ISSUE IN PARSING STRING (HITS DEFAULT DATE.DISTANTPAST)
```

Comment:
Resolve whatever this issue is referring to. Also, check out some of the Time.swift functions to see if we possibly already do something like this